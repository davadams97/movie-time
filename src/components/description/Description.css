.description {
    margin: 0 10px 0 10px;
    display: flex;
    flex-direction: column;
}

.date {
    font-weight: bolder;
    font-size: x-large;
}

.rating {
    margin-top: auto;
}

#btn {
    padding: 20px;
    margin: 4px;
    margin-top: auto;
}
