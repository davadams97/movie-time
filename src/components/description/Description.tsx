import React, { Component } from 'react';
import './Description.css';
import { IDescriptionProps } from './Description.types';

export class Description extends Component<IDescriptionProps, any> {
    render(): React.ReactElement {
        const { totalRating, buttonText, movieDescription } = this.props;

        return (
            <div data-testid="desc" className="description">
                <div>
                    <span data-testid="date" className="date">
                        {movieDescription?.release_date?.split('-')[0] ?? ''}
                    </span>
                </div>
                <div>
                    <em>{movieDescription?.runtime} mins.</em>
                </div>
                <div className="rating">
                    <span>
                        <strong data-testid="rating">
                            {movieDescription?.vote_average}/{totalRating}
                        </strong>
                    </span>
                </div>
                <button data-testid="button" id="btn">
                    {buttonText}
                </button>
            </div>
        );
    }
}
