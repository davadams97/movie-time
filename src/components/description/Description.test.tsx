import * as React from 'react';
import { Description } from './Description';
import { render, screen, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';

afterEach(() => cleanup());

test('renders the content correctly', () => {
    const { totalRating, buttonText, movieDescription } = {
        totalRating: 10,
        buttonText: 'Test',
        movieDescription: null,
    };
    render(<Description totalRating={totalRating} buttonText={buttonText} movieDescription={movieDescription} />);
    const element = screen.getByTestId('desc');
    expect(element).toBeInTheDocument();
});

test('should show rating correctly', () => {
    const { totalRating, buttonText, movieDescription } = {
        totalRating: 10,
        buttonText: 'Test',
        movieDescription: { vote_average: 4.6 },
    };
    render(<Description totalRating={totalRating} buttonText={buttonText} movieDescription={movieDescription} />);
    const element = screen.getByTestId('rating');
    expect(element).toHaveTextContent('4.6/10');
});

test('should show button text correctly', () => {
    const { totalRating, buttonText, movieDescription } = {
        totalRating: 7.5,
        buttonText: 'Click here!',
        movieDescription: null,
    };
    render(<Description totalRating={totalRating} buttonText={buttonText} movieDescription={movieDescription} />);
    const element = screen.getByTestId('button');
    expect(element).toHaveTextContent('Click here!');
});

test('should show release date correctly', () => {
    const { totalRating, buttonText, movieDescription } = {
        totalRating: 7.5,
        buttonText: 'Click here!',
        movieDescription: { release_date: '2021-10-01' },
    };
    render(<Description totalRating={totalRating} buttonText={buttonText} movieDescription={movieDescription} />);
    const element = screen.getByTestId('date');
    expect(element).toHaveTextContent('2021');
});

test('matches snapshot', () => {
    const { totalRating, buttonText, movieDescription } = {
        totalRating: 7.5,
        buttonText: 'Click here!',
        movieDescription: { release_date: '2021-10-01', vote_average: 7.2, runtime: 233 },
    };
    const tree = renderer
        .create(<Description totalRating={totalRating} buttonText={buttonText} movieDescription={movieDescription} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
