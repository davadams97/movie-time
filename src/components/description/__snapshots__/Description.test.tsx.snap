// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="description"
  data-testid="desc"
>
  <div>
    <span
      className="date"
      data-testid="date"
    >
      2021
    </span>
  </div>
  <div>
    <em>
      233
       mins.
    </em>
  </div>
  <div
    className="rating"
  >
    <span>
      <strong
        data-testid="rating"
      >
        7.2
        /
        7.5
      </strong>
    </span>
  </div>
  <button
    data-testid="button"
    id="btn"
  >
    Click here!
  </button>
</div>
`;
